base/Dockerfile

```diff
@@ -52,8 +52,10 @@ USER gitpod
 # use sudo so that user does not get sudo usage info on (the first) login
 RUN sudo echo "Running 'sudo' for Gitpod: success" && \
     # create .bashrc.d folder and source it in the bashrc
-    mkdir /home/gitpod/.bashrc.d && \
-    (echo; echo "for i in \$(ls \$HOME/.bashrc.d/*); do source \$i; done"; echo) >> /home/gitpod/.bashrc
+    mkdir -p /home/gitpod/.bashrc.d && \
+    if [ "$(ls -A $HOME/.bashrc.d/)" ]; then
```

Comment:
At this stage `/home/gitpod/.bashrc.d` will always be empty when the check is done (the `if` statement is done at build time not runtime).

the check should be done in `/home/gitpod/.bashrc` (runtime). Thus, the if statement should be with the for loop.

Reply:
Untested solution:

```
    (echo; echo "for i in \$(ls -A \$HOME/.bashrc.d/); do source \$HOME/.bashrc.d/\$i; done"; echo) >> /home/gitpod/.bashrc
```

The idea here is that `ls -A` on the directory itself comes empty if there are no files, and print the files (filename only) otherwise.

As `ls -A` prints only the filename the path needs amendment on the source line later.